### PR TITLE
refactor(pie-textarea): DSW-2469 replace label with form-label component

### PR DESCRIPTION
**Pull request by raoufswe**
## Describe your changes (can list changeset entries if preferable)

[Removed] - remove the `label` prop and use the pie-form-label component instead.

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.

### Reviewer 1 - @fernandofranca 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 @leksaBoiko 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

**Comment by leksaBoiko** [comment]
I noticed that the `maxLength` prop(in this [PR](https://github.com/justeattakeaway/pie/pull/2050)) was removed from the textarea component. 

Would it be better to remove section [Character counter](https://pr2054-docs.pie.design/components/textarea/#character-counter) as well?

File: .changeset/tiny-books-reply.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+"@justeattakeaway/pie-textarea": minor
+"pie-storybook": minor
+"pie-docs": minor
+---
+
+[Removed] - remove the `label` prop and use the pie-form-label component instead.
```
Comment (leksaBoiko) [inline .changeset/tiny-books-reply.md:7]:
```suggestion
[Updated] - remove the `label` prop and use the `pie-form-label` component instead.
```

**Review by leksaBoiko — DISMISSED** [review]

File: apps/pie-storybook/stories/pie-textarea.stories.ts [hunk 2]
```diff
@@ -225,15 +217,22 @@ const ExampleFormTemplate: TemplateFunction<TextareaProps> = () => html`
...
 
+const WithLabelTemplate: TemplateFunction<TextareaProps> = (props: TextareaProps) => html`
+        <p>Please note, the label is a separate component. See <pie-link href="/?path=/story/form-label">pie-form-label</pie-link>.</p>
+        <pie-form-label for="${ifDefined(props.name)}">Label</pie-form-label>
+        ${Template(props)}
+    `;
+
 const CreateTextareaStory = createStory<TextareaProps>(Template, defaultArgs);
 const CreateTextareaStoryWithForm = createStory<TextareaProps>(ExampleFormTemplate, defaultArgs);
+const createStoryWithLabel = (props: TextareaProps) => createStory<TextareaProps>(WithLabelTemplate, props);
```
Comment (fernandofranca) [inline apps/pie-storybook/stories/pie-textarea.stories.ts:228]: Should this follow the naming of the other stories, as `CreateStoryWithLabel`?

**Review by fernandofranca — APPROVED** [review]

File: apps/pie-storybook/stories/pie-textarea.stories.ts (hunk 2 above)
Comment (raoufswe) [reply apps/pie-storybook/stories/pie-textarea.stories.ts]: updated

**Review by leksaBoiko — APPROVED** [review]

File: apps/pie-storybook/stories/pie-textarea.stories.ts [hunk 3]
```diff
@@ -225,15 +217,22 @@ const ExampleFormTemplate: TemplateFunction<TextareaProps> = () => html`
...
 const CreateTextareaStoryWithForm = createStory<TextareaProps>(ExampleFormTemplate, defaultArgs);
+const createTextareaStoryWithLabel = (props: TextareaProps) => createStory<TextareaProps>(WithLabelTemplate, props);
 
 export const Default = CreateTextareaStory({}, {
     argTypes: {
         defaultValue: { table: { readonly: true }, description: 'This prop only works when the textarea is inside a form. To interact with this, view the Example Form story.' },
     },
 });
-
+export const WithLabel = createTextareaStoryWithLabel(defaultArgs)();
```
Comment (fernandofranca) [inline apps/pie-storybook/stories/pie-textarea.stories.ts:235]: Nit: Shouldn't `createTextareaStoryWithLabel` start with uppercase as `CreateTextareaStory` does?

**Review by ashleynolan — APPROVED** [review]

File: apps/pie-docs/src/components/textarea/overview/overview.md [hunk 4]
```diff
@@ -294,7 +294,7 @@ The character limit and counter is used to let users know how long their entry c
     {% contentPageImage {
       src: "../../../assets/img/components/textarea/character-counter.svg",
       width: "256px",
-      alt: "A textarea component with a label, and text inside it. The character counter on the top right of the component shows fifty out of fifty characters."
+      alt: "A textarea component with a label and text inside it. The character counter at the top right shows 50 out of 50 characters. At the bottom, assistive text in an error state indicates that the character limit has been exceeded by 43 characters."
```
Comment (xander-marjoram) [inline apps/pie-docs/src/components/textarea/overview/overview.md:297]:
```suggestion
      alt: "A textarea component with a label and text inside it. The character counter at the top right shows 93 out of 50 characters. At the bottom, assistive text in an error state indicates that the character limit has been exceeded by 43 characters."
```

**Comment by xander-marjoram** [comment]
I think the component status should be updated from "Planned" to "Alpha"

File: apps/pie-docs/src/components/textarea/overview/overview.md (hunk 4 above)
Comment (raoufswe) [reply apps/pie-docs/src/components/textarea/overview/overview.md:297]: updated

**Review by xander-marjoram — APPROVED** [review]

**Review by leksaBoiko — APPROVED** [review]

**Pull request merged**